src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -647,27 +666,19 @@ impl WorkunitStore {
   fn cancel_workunit(&self, workunit: Workunit) {
     workunit.log_workunit_state(true);
     self
-      .heavy_hitters_data
-      .msg_tx
-      .lock()
+      .sender
       .send(StoreMsg::Canceled(workunit.span_id))
-      .unwrap();
+      .unwrap_or_else(|_| panic!("Receivers are static, and should always be present."));
   }
 
   fn complete_workunit_impl(&self, mut workunit: Workunit, end_time: SystemTime) {
     let span_id = workunit.span_id;
     let new_metadata = Some(workunit.metadata.clone());
 
-    let tx = self.streaming_workunit_data.msg_tx.lock();
```

Comment:
This was a bug: the lock on the channel was being held for the rest of the method, rather than only long enough to send the message.